データベース設計

YUICHIROONUKI opened the pull request:
# WHAT
データベースに必要なテーブルとカラム、型、制約、アソシエーションをREADME.mdにマークダウン方式で記述しました。

#WHY
サービス開発前に設計することによりサービスが複雑になった場合でも効率的にデータ操作を行う事ができるため。

YUICHIROONUKI committed:
Edit README.md

YUICHIROONUKI committed:
Edit README.md again

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
-Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|name|string| |
```
null: false, index, uniqueをつけましょう

YUICHIROONUKI replied:
指摘箇所修正しました。
nameカラムのOptionsにnull: false, add_index, unique: trueを追記

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|password|integer|null: false|
 
-* Ruby version
+### Association
```
memersともアソシエーションを組んでください

YUICHIROONUKI replied:
membersとのアソシエーション
has_many :members 
を追記しました。

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+|name|string|null: false|
 
-* Database creation
+### Association
```
memersともアソシエーションを組んでください

YUICHIROONUKI replied:
membersとのアソシエーション
has_many :members
をgroupsテーブルに追記しました。

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+- has_many :users, through: :members
 
-* Database initialization
+## text_messagesテーブル
```
messagesで大丈夫です

YUICHIROONUKI replied:
text_messageからmessagesにテーブル名を変更しました。

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
-* How to run the test suite
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
```
画像も送れるようにしましょう

YUICHIROONUKI replied:
Imageカラムをmessageテーブルに追記しました

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,58 @@
+- belongs_to :group
 
-* Deployment instructions
+## photo_messagesテーブル
```
messagesテーブルにカラムを持たせれば大丈夫です

YUICHIROONUKI replied:
messageテーブルにimageカラムを追加したため必要なくなったphoto_messagesテーブルを削除した

YUICHIROONUKI committed:
Add name.Options

UsersテーブルのnameカラムのOptionsを変更した。
この編集によりユーザー名がnilでは保存されず、また同じユーザー名もデータベースに保存する事ができない。
インデックスを貼ることによりユーザー名のデータ検索を高速化した

YUICHIROONUKI committed:
Add users_Association

membersテーブルとのアソシエーションを追記した

YUICHIROONUKI committed:
Add groups_Association

Membersテーブルとのアソシエーションをgroupsテーブルに追記しました

YUICHIROONUKI committed:
Edit table_nameTO messages

テーブル名をtext_messagesからmessagesに変更しました。

YUICHIROONUKI committed:
Add image_column to messages_table

Imageカラムをmessageテーブルに追記しました

YUICHIROONUKI committed:
Delete photo_messages

Messageテーブルにimageカラムを追加したので必要のなくなったphoto_message テーブルを削除した

YUICHIROONUKI commented:
ご指摘ありがとうございます。

nameのoptionsを指定していなかった理由は、新規登録の際に名前がなくても登録できたため指定しませんでした。
また、messageテーブルのbodyカラムとimageカラムにnull: falseをつけると例えば画像だけを投稿するときにbodyカラムが空になり、データベースに保存できなくなると思ったのですが、いかがでしょうか？

TOMARUMARU reviewed (commented):
> nameのoptionsを指定していなかった理由は、新規登録の際に名前がなくても登録できたため指定しませんでした。

むむ、そんなはずはないです..
https://chat-space-proto.herokuapp.com/users/sign_up

> messageテーブルのbodyカラムとimageカラムにnull: falseをつけると例えば画像だけを投稿するときにbodyカラムが空になり、データベースに保存できなくなると思ったのですが、いかがでしょうか？

おっしゃる通りですね。body及びimageカラムはnullを許容してください

TOMARUMARU reviewed (commented):
> nameのoptionsを指定していなかった理由は、新規登録の際に名前がなくても登録できたため指定しませんでした。

むむ、そんなはずはないです..
https://chat-space-proto.herokuapp.com/users/sign_up

> messageテーブルのbodyカラムとimageカラムにnull: falseをつけると例えば画像だけを投稿するときにbodyカラムが空になり、データベースに保存できなくなると思ったのですが、いかがでしょうか？

おっしゃる通りですね。body及びimageカラムはnullを許容してください

YUICHIROONUKI committed:
Delete messages_options_null

画像だけを投稿した際にデータベースへの保存ができなくなるためmessageテーブルのbodyカラム、imageカラムのnull:
falseを削除した。

YUICHIROONUKI commented:
やはり名前がなくても登録できてしまいます。
ただ私もユーザー名は入力すべきであると思うのでこのまま進みたいと思います。

messageテーブルのbodyカラムとimageカラムのnullを許容しました。
問題なければマージしたいと思います。

よろしくお願いいたします。

TOMARUMARU reviewed (approved):
LGTMです(先ほど二回コメントを送ってしまったようで失礼いたしました...)
